### PR TITLE
docs(core): react-monorepo tutorial

### PR DESCRIPTION
**Pull request by isaacplmann**
Creates a [react-monorepo tutorial](https://nx-dev-git-fork-isaacplmann-docs-react-monorepo-tutorial-nrwl.vercel.app/getting-started/tutorials/react-monorepo-tutorial)

### CODE REVIEW

File: docs/shared/react-tutorial/react-monorepo.md [hunk 1]
```diff
@@ -0,0 +1,827 @@
...
+✓ built in 496ms
+
+ ——————————————————————————————————————————————————————————————————————————————————————————————————————————
+
+ >  NX   Successfully ran target build for project reactutorial (1s)
+```
+
+All the required files will be placed in the `dist/myreactapp` folder and can be deployed to your favorite hosting provider.
+
+## Visualizing your Project Structure
```
Comment (juristr) [inline docs/shared/react-tutorial/react-monorepo.md:567]: I would show the entire monorepo. THe graph right now only shows the `myreactapp` but doesn't show the `second-app`. I'd have at least a structure where we have a monorepo with these two apps and some libs that are just used by one app and some by both (e.g. the shared-ui for instance could be used by both). 

Or if we have these two `orders` and `products` we could have a second application `vendors` or `admin` that also imports `products` as a hypothetical app which manages the products on this store.

File: docs/shared/react-tutorial/react-monorepo.md [hunk 2]
```diff
@@ -0,0 +1,827 @@
...
+      <Route path="/products" element={<ProductList />}></Route>
+      <Route path="/orders" element={<OrderList />}></Route>
+    </Routes>
+  );
+}
+
+export default App;
+```
+
+## Building the App for Deployment
```
Comment (juristr) [inline docs/shared/react-tutorial/react-monorepo.md:546]: I think we should explain more here, like how we handle deployments for both apps. That is..

- deploy both individually => running the build for both individually
- just run `npx nx run-many -t build -p myreactapp second-app`
- use `affected:build`

for the "affected" to work here, we need to have shown it before. I forgot adding a section for that on the Linear issue, but I'd add that after the "Testing and Linting...." where we show running multiple commands and caching. I'd have a section right after that shows how to only run it for a subset of the monorepo, show the affected part n the graph etc...

File: docs/shared/react-tutorial/react-monorepo.md [hunk 3]
```diff
@@ -0,0 +1,827 @@
...
+         - nx run products:lint
+```
+
+If you have the ESLint plugin installed in your IDE you should immediately see an error:
+
+![ESLint module boundary error](/shared/images/tutorial-react-standalone/react-standalone-module-boundaries.png)
+
+Learn more about how to [enforce module boundaries](/core-features/enforce-module-boundaries).
+
+## Next Steps
```
Comment (juristr) [inline docs/shared/react-tutorial/react-monorepo.md:810]: on the linear issue, I also had a section "Setting up CI". Any reason that was skipped here?

**Review by juristr — CHANGES_REQUESTED** [review]

File: docs/shared/react-tutorial/react-monorepo.md [hunk 4]
```diff
@@ -0,0 +1,1039 @@
...
+
+Here's some more things you can dive into next:
+
+- Learn more about the [underlying mental model of Nx](/concepts/mental-model)
+- Learn how to [migrate your CRA app to Nx](/recipes/react/migration-cra)
+- [Learn how to setup Tailwind](/recipes/react/using-tailwind-css-in-react)
+- [Setup Storybook for our shared UI library](/recipes/storybook/overview-react)
+- [Speed up CI: Run only tasks for project that got changed](/core-features/run-tasks#run-tasks-affected-by-a-pr)]
```
Comment (juristr) [inline docs/shared/react-tutorial/react-monorepo.md:1030]: there's a `]` too much, same for the next line

File: docs/shared/react-tutorial/react-monorepo.md [hunk 5]
```diff
@@ -0,0 +1,1039 @@
...
+
+The `"dependsOn": "build"` setting tells Nx to make sure that the project's `build` task has been run successfully before the `deploy` task.
+
+With the `deploy` tasks defined, you can deploy a single application with `nx deploy react-store` or deploy any applications affected by the current changes with:
+
+```shell
+nx affected -t deploy
+```
+
+### Caching
```
Comment (juristr) [inline docs/shared/react-tutorial/react-monorepo.md:808]: I think this accidentally finished up in the wrong section. Should have been in the "Testing and Linting ..." section.

**Review by juristr — CHANGES_REQUESTED** [review]
LGTM, just some minor things. 
Can you also include it in the main intro page. There should already be a tile to uncomment for the React monorepo tutorial; the link just needs to be adjusted afair

Thanks!

**Comment by isaacplmann** [comment]
I'll wait to add the tutorial to the intro until we have the other 2 tutorials.  It looks bad adding just one on its own row
Screenshot 2023-09-13 at 11 58 08 AM

**Review by juristr — APPROVED** [review]

**Pull request merged**